paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/procedure/ExpireSnapshotsProcedure.java

```diff
@@ -85,9 +88,21 @@ public InternalRow[] call(InternalRow args) {
                     if (retainMin != null) {
                         builder.snapshotRetainMin(retainMin);
                     }
-                    if (olderThanMills != null) {
-                        builder.snapshotTimeRetain(
-                                Duration.ofMillis(System.currentTimeMillis() - olderThanMills));
+                    if (!StringUtils.isNullOrWhitespaceOnly(olderThanStr)) {
+                        long olderThanMills;
+                        // forward compatibility for timestamp type
```

Comment:
How to forward compatibility for timestamp type? It is a string instead of a timestamp.

Reply:
In default, the value of timestamp type is like "1724664000000", it's actually a string that contains numerics. And get it by 
`Long olderThanMills = args.isNullAt(3) ? null : args.getLong(3) / 1000;`

Reply:
`change param older_than type from TimeStampType to StringType, TimeStampType is not easy to understand and use.`

Is this related? Can you show the use cases? How to improve the usage.

Reply:
got, has only retain StringType